checker/check_result.go

```diff
@@ -56,6 +56,48 @@ type DetailLogger interface {
 	Debug(desc string, args ...interface{})
 }
 
+// FileType is the type of a file.
+type FileType int
+
+const (
+	// FileTypeNone is a default, unspecified type.
+	FileTypeNone FileType = iota
+	// FileTypeSource is for source code.
+	FileTypeSource
+	// FileTypeBinary is fir binary files.
+	FileTypeBinary
+	// FileTypeText is text.
+	FileTypeText
+	// FileTypeURL is for URLs.
+	FileTypeURL
+)
+
+// LogMessage is a structure that encapsulates detail's information.
+// This allows updating the definition easily.
+//nolint
+type LogMessage struct {
+	Text    string   // A short string explaining why the detail was recorded/logged.
+	Path    string   // Fullpath to the file.
+	Type    FileType // Type of file.
+	Offset  int      // Offset in the file of Path (line for source/text files).
+	Snippet string   // Snippet of code.
+}
+
+// CheckDetail3 contains information for each detail.
+// UPGRADEv3: rename to DetailLogger.
+type CheckDetail3 struct {
```

Comment:
Rather than introduce yet another similar but different structure, can we just re-use a single structure? 

either extending the original `CheckDetail` without breaking compatibility

```go
type CheckDetail struct {
  Msg string
  Type DetailType
  Details struct {
    Path string
    Type FileType
    Offset int
    Snippet string
  }
}
```

Or just migrate it to what you have here, and have the non-detailed logger store `Msg.Text`  and keep everything else empty.

If you later need export this in a different JSON format, we can convert it to the structure structure then.

Reply:
sg. done. ptal